aptos-move/framework/aptos-framework/sources/storage_gas.move

```diff
@@ -181,18 +192,31 @@ module aptos_framework::storage_gas {
 
     fun validate_usage_config(config: &UsageGasConfig) {
         assert!(config.target_usage > 0, error::invalid_argument(EZERO_TARGET_USAGE));
-        assert!(config.target_usage <= 18446744073709551615 / BASIS_POINT_DENOMINATION, error::invalid_argument(ETARGET_USAGE_TOO_BIG));
+        assert!(config.target_usage <= MAX_U64 / BASIS_POINT_DENOMINATION, error::invalid_argument(ETARGET_USAGE_TOO_BIG));
         validate_curve(&config.read_curve);
         validate_curve(&config.create_curve);
         validate_curve(&config.write_curve);
     }
 
     fun validate_curve(curve: &GasCurve) {
         assert!(curve.max_gas >= curve.min_gas, error::invalid_argument(EINVALID_GAS_RANGE));
+        assert!(curve.max_gas <= MAX_U64 / BASIS_POINT_DENOMINATION, error::invalid_argument(EINVALID_GAS_RANGE));
         let points = &curve.points;
         let len = vector::length(points);
+        spec {
+            assume len < MAX_U64;
```

Comment:
Otherwise, Prover complains because if `len == MAX_U64`, `i = i + 1` can abort at the end of the loop.
We know/assume that the vector can't be that large.